source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -153,15 +153,42 @@ def expand(self, unit):
 		else:
 			return super(consoleUIATextInfo, self).expand(unit)
 
-	def _get_isCollapsed(self):
+	def compareEndPoints(self, other, which):
 		"""Works around a UIA bug on Windows 10 1803 and later."""
 		# Even when a console textRange's start and end have been moved to the
 		# same position, the console incorrectly reports the end as being
 		# past the start.
-		# Therefore to decide if the textRange is collapsed,
-		# Check if it has no text.
+		# Compare to the start (not the end) when collapsed.
+		selfEndPoint, otherEndPoint = which.split("To")
+		if selfEndPoint == "end" and not self.hasNoText:
+			selfEndPoint = "start"
+		if otherEndPoint == "End" and not other.hasNoText:
+			otherEndPoint = "Start"
+		which = f"{selfEndPoint}To{otherEndPoint}"
+		return super().compareEndPoints(other, which=which)
+
+	def setEndPoint(self, other, which):
+		"""Works around a UIA bug on Windows 10 1803 and later."""
+		# Even when a console textRange's start and end have been moved to the
+		# same position, the console incorrectly reports the end as being
+		# past the start.
+		selfEndPoint, otherEndPoint = which.split("To")
+		# In this case, there is no need to check selfEndPoint
+		# since it is about to be overwritten in the super call.
```

Comment:
```suggestion
		# since the point of this method is to change its text range, modifying the "end" endpoint of a collapsed
		# textrange is fine.
```